src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -5998,8 +5998,8 @@ void AuraEffect::HandlePeriodicHealthLeechAuraTick(Unit* target, Unit* caster) c
 
     caster->CalcAbsorbResist(target, GetSpellInfo()->GetSchoolMask(), DOT, damage, &absorb, &resist, m_spellInfo);
 
-    if (target->GetHealth() < damage)
-        damage = uint32(target->GetHealth());
+    if (target->GetMaxHealth() < damage)
+        damage = target->GetMaxHealth();
```

Comment:
I think that @Eliminationzx will need to answer on that question. From my aspect of looking as dev, this isn't very smart way because this could affect other things as well, at least in my opinion.